packages/app-desktop/gui/MainScreen/commands/toggleEditorFullScreen.ts

```diff
@@ -0,0 +1,65 @@
+import {
+	CommandContext,
+	CommandDeclaration,
+	CommandRuntime,
+} from '@joplin/lib/services/CommandService';
+import { _ } from '@joplin/lib/locale';
+import setLayoutItemProps from '../../ResizableLayout/utils/setLayoutItemProps';
+import layoutItemProp from '../../ResizableLayout/utils/layoutItemProp';
+import { AppState } from '../../../app.reducer';
+
+export const declaration: CommandDeclaration = {
+	name: 'toggleEditorFullScreen',
+	label: () => _('toggle full screen'),
+	iconName: 'fas fa-expand',
+};
+
+export const runtime = (): CommandRuntime => {
+	return {
+		execute: async (context: CommandContext) => {
+			const layout = (context.state as AppState).mainLayout;
+
+			let isEditorFullScreen = layoutItemProp(layout, 'editor', 'isFullScreen');
+
+			if (isEditorFullScreen === undefined) {
+				const isNoteListVisible = layoutItemProp(layout, 'noteList', 'visible');
+				const isSidebarVisible = layoutItemProp(layout, 'sideBar', 'visible');
+
+				isEditorFullScreen = !isNoteListVisible && !isSidebarVisible;
+			}
+
+			let newLayout = layout;
+
+			if (isEditorFullScreen) {
+				newLayout = setLayoutItemProps(newLayout, 'sideBar', {
+					visible: true,
+				});
+
+				newLayout = setLayoutItemProps(newLayout, 'noteList', {
+					visible: true,
+				});
+			} else {
+				newLayout = setLayoutItemProps(newLayout, 'sideBar', {
+					visible: false,
+				});
+				newLayout = setLayoutItemProps(newLayout, 'noteList', {
+					visible: false,
+				});
+			}
+
+			newLayout = setLayoutItemProps(newLayout, 'editor', {
+				isFullScreen: !isEditorFullScreen,
+			});
+
+			// Toggling the sidebar will affect the size of most other on-screen components.
+			// Dispatching a window resize event is a bit of a hack, but it ensures that any
+			// component that watches for resizes will be accurately notified
+			window.dispatchEvent(new Event('resize'));
+
+			context.dispatch({
+				type: 'MAIN_LAYOUT_SET',
+				value: newLayout,
+			});
+		},
```

Comment:
Thanks for your review, I don't know the global code very well, can you give an example how to call the existing toggleNoteList and toggleSideBar commands here?

As far as my personal experience is concerned, when I am writing, I habitually want to make the editor full screen so that I can focus on writing. At present, I can only achieve it by collapsing the sidebar and note list, which is not very convenient.

As for the problem you mentioned, I think it can be optimized in this way. Before triggering the full screen of the editor, record the current state of the sidebar and note list, and then restore it after exiting the full screen.

Reply:
1. Trigger the sidebar and note list through command calls;
2. Record the status of the sidebar and note list of the user interface, and restore it when exiting the full screen of the editor.